src/abstractprobprog.jl

```diff
@@ -60,3 +60,19 @@ m = decondition(condition(m, obs))
 should hold for generative models `m` and arbitrary `obs`.
 """
 function condition end
+
+""" 
+    AbstractContext
+
+Common base type for Contexts used for dispatching.
+"""
+abstract type AbstractContext end
+
+
+""" 
+    evaluate!!
+
+Sample from the `model` using the `sampler` with random number generator `rng` and the
```

Comment:
This doc string is no longer accurate. `evaluate!` is meant as a general API for various kinds of model operations, e.g. sampling from the prior, evaluating various log densities (prior, loglike and logjoint).